source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1415,6 +1415,32 @@ TEST(NiFakeServiceTests, NiFakeService_GetAnIviDanceWithATwistArray_CallsGetAnIv
     EXPECT_EQ(response.actual_size(), expected_size);
 }
 
+TEST(NiFakeServiceTests, NiFakeService_GetArrayViUInt8WithEnum_CallsGetArrayViUInt8WithEnum)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  int array_len = 3;
+  // using string because in grpc, ViUInt8 array gets converted to string
+  std::string uint8_array = {nifake_grpc::Color::COLOR_RED, nifake_grpc::Color::COLOR_BLACK, nifake_grpc::Color::COLOR_BLUE};
+  nifake_grpc::Color enum_array[] = {nifake_grpc::Color::COLOR_RED, nifake_grpc::Color::COLOR_BLACK, nifake_grpc::Color::COLOR_BLUE};
+  EXPECT_CALL(library, GetArrayViUInt8WithEnum(kTestViSession, array_len, _))
+      .WillOnce(DoAll(SetArrayArgument<2>(enum_array, enum_array + array_len), Return(kDriverSuccess)));
```

Comment:
Split line like this for better readability. You also seem to have few extra spaces before `.WillOnce`. We add extra spaces per indent level.

```
    .WillOnce(
      DoAll(
        SetArrayArgument<2>(enum_array, enum_array + array_len), 
        Return(kDriverSuccess)));
```

Reply:
fixed